ot/lp/dmmot.py

```diff
@@ -0,0 +1,319 @@
+# -*- coding: utf-8 -*-
+"""
+d-MMOT solvers for optimal transport
+"""
+
+# Author: Ronak Mehta <ronakrm@cs.wisc.edu>
+#         Xizheng Yu <xyu354@wisc.edu>
+#
+# License: MIT License
+
+import numpy as np
+from ..backend import get_backend
+
+
+def dist_monge(i):
+    r"""
+    A tensor :math:c is Monge if for all valid :math:i_1, \ldots i_d and
+    :math:j_1, \ldots, j_d,
+
+    .. math::
+        c(s_1, \ldots, s_d) + c(t_1, \ldots t_d) \leq c(i_1, \ldots i_d) +
+        c(j_1, \ldots, j_d)
+
+    where :math:s_k = \min(i_k, j_k) and :math:t_k = \max(i_k, j_k).
+
+    Our focus is on a specific cost, which is known to be Monge:
+
+    .. math::
+        c(i_1,i_2,\ldots,i_d) = \max{i_k:k\in[d]} - \min{i_k:k\in[d]}.
+
+    When :math:d=2, this cost reduces to :math:c(i_1,i_2)=|i_1-i_2|,
+    which agrees with the classical EMD cost. This choice of :math:c is called
+    the generalized EMD cost.
+
+    Parameters
+    ----------
+    i : list
+        The list for which the generalized EMD cost is to be computed.
+
+    Returns
+    -------
+    cost : numeric value
+        The generalized EMD cost of the tensor.
+
+    References
+    ----------
+    .. [51] Jeffery Kline. Properties of the d-dimensional earth mover's
+        problem. Discrete Applied Mathematics, 265: 128-141, 2019.
+    .. [53] Wolfgang W. Bein, Peter Brucker, James K. Park, and Pramod K.
+        Pathak. A monge property for the d- dimensional transportation problem.
+        Discrete Applied Mathematics, 58(2):97-109, 1995. ISSN 0166-218X. doi:
+        https://doi.org/10.1016/0166-218X(93)E0121-E. URL
+        https://www.sciencedirect.com/ science/article/pii/0166218X93E0121E.
+        Workshop on Discrete Algoritms.
+    """
+
+    return max(i) - min(i)
+
+
+def discrete_mmot(A, verbose=False, log=False):
+    r"""
+    Compute the discrete multi-marginal optimal transport of distributions A.
```

Comment:
you should explain clearly ere that you suppose that the support of the distributions are supposed integers on the real line, this will be suggested by the new function name but it needs to be stated clearly.